### PR TITLE
Fix incorrect parsing of the ATCA FITS header

### LINKED ISSUE #1375: incorrect trimmed velocity header of ATCA FITS images
**Describe the bug**
This is related to https://github.com/CARTAvis/carta-frontend/issues/2357. 
The frontend receives incorrect headers of velocity(CTYPE3) from ATCA FITS images.
The original value of CTYPE3 in the FITS headers is **'VELO-XXX'** (eg. VELO-HEL, VELO-LSR), but the OpenFileAck message shows **'VELO'** in that key..

VELO-HEL turns to VELO
截圖 2024-05-02 下午2 38 03
截圖 2024-05-02 下午2 48 59

VELO-LSR turns to VELO
截圖 2024-05-02 下午3 09 36
截圖 2024-05-02 下午2 58 14

**To Reproduce**
1. download the test images here: 
https://drive.google.com/file/d/1MrseMvR3j9YVeLk_4bwCmDjvIaPHUtye/view?usp=sharing (CTYPE3 = 'VELO-HEL')
https://drive.google.com/file/d/1ynmyPlgEJolb_1D8m5tWNapXIGMwmUHO/view?usp=sharing (CTYPE3 = 'VELO-LSR')

2. load it and check the log of OpenFileAck(or File header), and we will see the incorrect header.

**Expected behavior**
The CTYPE3 in the OpenFileAck message should be the same as the one in the FITS header.

**Screenshots or videos**
If applicable, add screenshots/videos to help explain your problem.

**Platform info (please complete the following information):**
 - OS [e.g. macOS Monterey]: macOS Sonoma
 - Browser [e.g. chrome, safari, electron app]: Chrome
 - Browser version [e.g. 22]: 124
 - Backend branch [e.g. dev, v3b2 release]: dev
 - Frontend branch [e.g. dev, v3b2 release]: dev

**Additional context**

### PR DESCRIPTION
**Pull request by markccchiang**
**Description**

* What is implemented or fixed? Mention the linked issue(s), if any.
Closes #1375.  

* How does this PR solve the issue? Give a brief summary.
The value of SPECSYS from the ATCA FITS header is already provided. So we don't need to parse it from the suffix of VELO-XXX (the string after '-'), which would also result in the failure rendering of the coordinate system.

* Are there any companion PRs (frontend, protobuf)?
No.

* Is there anything else that testers should know (e.g. exactly how to reproduce the issue)?
The coordinate system for ATCA FITS images can be successfully rendered on the frontend.

**Checklist**

- [x] changelog updated ~/ no changelog update needed~
- [x] e2e test passing / corresponding fix added / new e2e test created
- [ ] ICD test passing / corresponding fix added / new ICD test created
- [x] ~protobuf updated to the latest dev commit /~ no protobuf update needed
- [x] ~protobuf version bumped /~ protobuf version not bumped
- [ ] added reviewers and assignee
- [ ] GitHub Project estimate added

### CODE REVIEW

**Comment by pford** [comment]
Should the original ctype value VELO-XXX also be unchanged?  The issue requested the original header.  I think I wrote this code, not sure why I changed the value but maybe a casacore issue at the time.

Also if the SPECSYS header is read before the CTYPE header, then the specsys string will be replaced with the CTYPE value, which is not what we want.

Starting at line 319 in this branch FileExtInfoLoader.cc:

            } else if (value.startsWith("VELO-") && specsys.empty()) {   <== only if not set?
                specsys = value.after('-');
                value = "VELO";     <=== remove this line?
            }

**Comment by markccchiang** [comment]
I noticed that if I do not remove the string after VELO (from the original VELO-XXX), the x-label of the spectral profile plot can not be shown normally. Like the following snapshot:

截圖 2024-05-16 上午10 11 05

with "CTYPE3 = VELO" the x-label is normally shown: 

截圖 2024-05-16 上午10 12 33

Not sure if this is the AST problem, or just the frontend code issue(?) @kswang1029 @TienHao do you have comments on it?

**Comment by TienHao** [comment]
> Not sure if this is the AST problem, or just the frontend code issue(?) @kswang1029 @TienHao do you have comments on it?

Yeah, the frontend code needs to be modified for VELO-XXX. It does not give spectral type name for VELO-XXX.

@kswang1029, do VELO-XXX have the same spectral type name and unit just like VELO? or they should have other names?

**Comment by kswang1029** [comment]
> > Not sure if this is the AST problem, or just the frontend code issue(?) @kswang1029 @TienHao do you have comments on it?
> 
> 
> 
> Yeah, the frontend code needs to be modified for VELO-XXX. It does not give spectral type name for VELO-XXX.
> 
> 
> 
> @kswang1029, do VELO-XXX have the same spectral type name and unit just like VELO? or they should have other names?
> 
> 

Yes just like VELO. The VELO-XXX is a (old?) way to encode the spectral reference frame. We still refer CUNIT3 for the unit.

**Review by pford — APPROVED** [review]

**Comment by pford** [comment]
I approved the code review, and will leave it up to the others whether to keep the change to "VELO" or show the actual CTYPE header value.  It sounds like the suffix is outdated, does not add any information since SPECSYS is defined, and requires a workaround for the spectral profile.

**Comment by TienHao** [comment]
Since the frontend is working with CTYPE3 = VELO in the current PR(no VELO-XXX is sent through headers), I think there is no need to change in the frontend for the spectral type name now.

**Review by confluence — APPROVED** [review]

**Comment by YuHsuan-Hwang** [comment]
Does this PR still need frontend changes?

**Comment by kswang1029** [comment]
> Does this PR still need frontend changes?

I will check and report back.

**Comment by kswang1029** [comment]
> Does this PR still need frontend changes?

No, we do not need a corresponding frontend change. 👍

**Review by kswang1029 — APPROVED** [review]
issue fixed. no regression from e2e tests. corresponding frontend change is not needed.

**Comment by kswang1029** [comment]
@confluence ready to merge after fixing the changelog conflict.

**Pull request merged**
